src/plugins/midi-piano/midi-piano-display.js

```diff
@@ -0,0 +1,292 @@
+import axios from 'axios';
+import * as Tone from 'tone';
+import { Button } from 'antd';
+import midiPlayerNs from 'midi-player-js';
+import urlUtils from '../../utils/url-utils.js';
+import React, { useEffect, useRef } from 'react';
+import PianoComponent from './piano-component.js';
+import { QuestionCircleOutlined } from '@ant-design/icons';
+import ClientConfig from '../../bootstrap/client-config.js';
+import { useService } from '../../components/container-context.js';
+import { sectionDisplayProps } from '../../ui/default-prop-types.js';
+import PlayIcon from '../../components/icons/media-player/play-icon.js';
+import PauseIcon from '../../components/icons/media-player/pause-icon.js';
+
+export default function MidiPianoDisplay({ content }) {
+
+  const keys = useRef([]);
+  const player = useRef(null);
+  const sampler = useRef(null);
+  const midiData = useRef(null);
+  const midiAccessObj = useRef(null);
+  const samplerHasLoaded = useRef(false);
+  const { NOTES } = midiPlayerNs.Constants;
+  const { sourceType, sourceUrl, midiTrackTitle, noteRange } = content;
+  const hasMidiFile = sourceUrl !== '';
+  const hasMidiTrackTitle = midiTrackTitle !== '';
+  const clientConfig = useService(ClientConfig);
+  const src = urlUtils.getMidiUrl({ cdnRootUrl: clientConfig.cdnRootUrl, sourceType, sourceUrl });
+
+  const getNoteNameFromMidiValue = midiValue => {
+    return NOTES[midiValue];
+  };
+
+  const getEventTypefromMidiCommand = (command, velocity) => {
+    switch (command) {
+      case 144:
+        if (velocity > 0) {
+          return 'Note on';
+        }
+        return 'Note off';
+      case 128:
+        return 'Note off';
+      case 176:
+        return 'Sustain off';
```

Comment:
As far as I can see, this "Sustain off" value is never used, just "Note on" and "Note off", so you could just remove it.
You could extract the 2 values that are actually used into a const (and naming them a bit clearer):
```
const PIANO_EVENT = {
   notePressed: 'note-pressed',
   noteReleased: 'note-released',
   other: 'other'
}
```

Reply:
Yes, should have commented it out, because it will be used soon, so that when you play the piano with a midi device you can actually use the pedal.

Reply:
I thought this was used 🤔 
I'd recommend not having either unused or commented out code in your pull request and keeping it clean. Any new functionality you want to add later can then be built on top.

Reply:
I removed the "Sustain off" and kept the other two named the same just for now, still need to think about it based on how the plugin evolves. Thanks however for the suggestion, definitely open to it!